### DISCUSSION TITLE
Buddy Firmware Flash via USB on EL18 Not Working

### DISCUSSION THREAD

**Discussion by Setizar** [EdgeTX/edgetx#3219] [Q&A] [answered]
Hi,

1st timer EdgeTX user and looking forward to participating in the EdgeTX community.

I am trying to flash via USB my EL18 to V2.1.8 using Buddy and the browser interface but cannot identify the correct available device to use.  I do not see "STM32 Bootloader" device.  Only see two unknown devices and others from my notebook.  ie. 0x2A94:0x5009 and 0x3938:0x1031.  Advice is welcome please.

Darren

**Comment by pfeerick** [comment, marked as answer]
Are you putting the EL18 in DFU mode? (i.e. you've taken the neck strap attachment off, and had the concealed buttoned at the bottom of the back screw-hole pressed in when connecting the USB to the computer?)

**Reply by Setizar** [reply]
Hi Peter.  Thanks for the tip.  Yes i have the button depressed with the Tx powered off when connecting the USB to my notebook.  Still cant see the  "STM32 Bootloader" device.

**Reply by pfeerick** [reply]
Ok, it's not that simple 😆 

I would then try the ImpulseRC Driver Fixer from https://impulserc.com/pages/downloads if you are on Windows - there is a good chance the DFU driver isn't present, hence why it can't see it. 

The fallback method of flashing via the firmware and bootloader is also an option - whereby you place the firmware in the `FIRMWARE` folder of the radio SD card, and then you can update the bootloader via the Radio Settings -> SD Browser, and update the firmware via the bootloader (hold the trim hats inwards while powering on the radio, then choose write firmware and select the firmware file to write). 

But obviously, it would be nice to have the DFU mode working as it updates both for you in one go.

**Comment by Setizar** [comment]
Thanks Peter.  Seems Impulse RC Diver cannot detect my flight controller.  The process seems simple enough but no luck,  I will work through the manual process you describe and let you know how I go.  Thanks again for your kind support.

I do have two drives, D: and E: and both contain a "Firmware" folder.  Seems I may need to remove the SD card in order to locate the firmware update in the E: which I cant seem to do while in the Tx.    It has no impact if I locate the firmware update in the D:.

Secondly I will work to get the DFU driver active on my TX.

**Reply by pfeerick** [reply]
One drive should contain several folders (MODELS, RADIO, etc ), including a FIRMWARE folder (create it if it's missing for some reason). The other drive will have a FIRMWARE.BIN file, and perhaps one other file. Leave that drive and file alone.

**Comment by Setizar** [comment]
Noted and thanks for the clarification.  I will proceed accordingly.   :-)

**Reply by Setizar** [reply]
Success with the Flash thanks!  Now I am looking to add the Heli configuration profile which is not included in EdgeTX V2.8.1.  Is this module available to the EL18 please?

**Reply by pfeerick** [reply]
The heli tab is disabled by default, but you can download a firmware build of 2.8.1 with that present from here : https://github.com/pfeerick/lang-firmwares/releases/tag/release ... pick the `en` (English) version from `el18-heli-lang-v2.8.1.zip` (unless you also the text in different language, of course!).

**Comment by Setizar** [comment]
Thanks Peter.  All done including the Heli tab.  Appreciate your help.

**Reply by pfeerick** [reply]
Very welcome. Happy flying! :)

**Comment by Setizar** [comment]
Actually I am still a long way from using the EL18 in the field.  I cant even bind an Rx to it at this time.  The Elysium user guide is no help and I cant find reference to it in the Edge 2,8 User Guide.  Do we have an EL18 expert in this forum?

**Reply by pfeerick** [reply]
I've got one and have used it... so what's your problem. Need to start of which what receiver you're trying to bind - as some have a push button to bind, some have bind plug, and with ADSFHD3, you also need to set the correct... protocol variant? ... for the Rx (it's a bit like Frsky where it has long range, high speed, or more channel variants). Basically it is set the right variant, tell the transmitter to bind, and then plug the Rx power in while it is in bind mode (button held, bund plug) and then it should bind instantly.

**Comment by Setizar** [comment]
I know a little about the variants and the protocols used in traditional RC systems and binding with buttons and the bind plug, but when it comes to EdgeTX and the EL18 I think my problem knowing the Tx and selecting the correct one.  Also is the internal RF Tx suitable for normal operations or do we need to buy the externa RFl module to connect to various other Rx's?  I have tried multiple Rx's including Tunigy, Specktrum, Fly Sky & Oragne but none would bind.  I thought Fly Sky would bind simply but no luck.  Is there a video on Youtube that you recommend to walk me through the process of binding different Rx brands to the EL18.

**Comment by richardclli** [comment]
EL18 do not have multiprotocol build in, you need to use an external multiprotocol 4 in 1 in you want to use a lot of different branded receivers.  EL18 has an internal module that only support Flysky AFHDS3 receivers.

**Comment by Setizar** [comment]
Thanks.  As I suspected.  Can you tell me exactly what the part number is and if I need to buy a cradle and where to buy it please?  Do I need to buy a Fly Sky external module or can I buy another brand?  Maybe a link or supplier site would be very helpful if you can.

**Comment by gagarinlg** [comment]
look for IRangeX-IRX4-Plus or https://www.radiomasterrc.com/products/rm-4in1-module
and one of those, if it did not come with the radio
https://hobbymatehobby.com/en-de/products/flysky-el18-elysium-rc-transmitter-external-module-bay-adapter?variant=43819562762478

**Reply by Setizar** [reply]
thank you for the quick reply.  Both have now been ordered.  👍

**Reply by pfeerick** [reply]
I'm a bit late to the party, but for the EL18, you might actually find the IRangeX IRX4 **_Lite_** a better fit. That module adapter is available for Nano modules also, so you could use the RM4IN1 directly, or the Lite version of the iRange IRx4 and it wouldn't be quite as bulky as the JR module bay and adapter ;) Just another option.

**Comment by Setizar** [comment]
Thanks Peter.  Good to have options to get full functionality from this radio.  Sadly the factory is very short sighted by releasing a product that binds with only one of their own receivers and I fell into their trap.  :-)

**Reply by pfeerick** [reply]
The same thing would have happened if you had bought any of the Frsky transmitters, or an ELRS transmitter... the internal RF module will only talk the one protocol (i.e. Frsky or ELRS). It's only a radio that have an internal MPM (MultiProtocol Module) that would talk to the various different receivers (without using a external module). Just be glad you didn't get a radio that *doesn't* have the facility for an external module so that you can use it with whatever RF system you wish. ;)

**Comment by Setizar** [comment]
Is it the same for Radiomaster?  A mate in the club urged me to buy one.

**Reply by pfeerick** [reply]
It depends on the type you get. If you get a Radiomaster transmitter with internal ELRS, then it only talks ELRS. If you get a Radiomaster transmitter with internal MPM/MULTI, it then talks all the protocols the MPM talks.

**Comment by Setizar** [comment]
It is all about cost.  Thanks Peter.  I will let you know how I go when the chassis/cradle and RF Module arrive.
